cub/test/catch2_test_thread_scan_exclusive_partial.cu

```diff
@@ -0,0 +1,465 @@
+      break;
+    case 5:
+      thread_scan_exclusive_partial_kernel<5><<<1, 1>>>(d_in, d_out, scan_operator, valid_items, prefix, apply_prefix);
+      break;
+    case 6:
+      thread_scan_exclusive_partial_kernel<6><<<1, 1>>>(d_in, d_out, scan_operator, valid_items, prefix, apply_prefix);
+      break;
+    case 7:
+      thread_scan_exclusive_partial_kernel<7><<<1, 1>>>(d_in, d_out, scan_operator, valid_items, prefix, apply_prefix);
+      break;
+    case 8:
+      thread_scan_exclusive_partial_kernel<8><<<1, 1>>>(d_in, d_out, scan_operator, valid_items, prefix, apply_prefix);
+      break;
+    case 9:
+      thread_scan_exclusive_partial_kernel<9><<<1, 1>>>(d_in, d_out, scan_operator, valid_items, prefix, apply_prefix);
+      break;
+    case 10:
+      thread_scan_exclusive_partial_kernel<10><<<1, 1>>>(d_in, d_out, scan_operator, valid_items, prefix, apply_prefix);
+      break;
+    case 11:
+      thread_scan_exclusive_partial_kernel<11><<<1, 1>>>(d_in, d_out, scan_operator, valid_items, prefix, apply_prefix);
+      break;
+    case 12:
+      thread_scan_exclusive_partial_kernel<12><<<1, 1>>>(d_in, d_out, scan_operator, valid_items, prefix, apply_prefix);
+      break;
+    case 13:
+      thread_scan_exclusive_partial_kernel<13><<<1, 1>>>(d_in, d_out, scan_operator, valid_items, prefix, apply_prefix);
+      break;
+    case 14:
+      thread_scan_exclusive_partial_kernel<14><<<1, 1>>>(d_in, d_out, scan_operator, valid_items, prefix, apply_prefix);
+      break;
+    case 15:
+      thread_scan_exclusive_partial_kernel<15><<<1, 1>>>(d_in, d_out, scan_operator, valid_items, prefix, apply_prefix);
+      break;
+    case 16:
+      thread_scan_exclusive_partial_kernel<16><<<1, 1>>>(d_in, d_out, scan_operator, valid_items, prefix, apply_prefix);
+      break;
+    default:
+      FAIL("Unsupported number of items");
+  }
+  REQUIRE(cudaSuccess == cudaPeekAtLastError());
+  REQUIRE(cudaSuccess == cudaDeviceSynchronize());
+}
+
+/***********************************************************************************************************************
+ * Test cases
+ **********************************************************************************************************************/
+
+C2H_TEST("ThreadScanExclusive Integral Type Tests", "[scan][thread]", integral_type_list, cub_operator_integral_list)
+{
+  using params                     = params_t<TestType>;
+  using value_t                    = typename params::item_t;
+  using output_t                   = typename params::output_t;
+  using op_t                       = c2h::get<1, TestType>;
+  using accum_t                    = ::cuda::std::__accumulator_t<op_t, value_t>;
+  using dist_param                 = dist_interval<value_t, op_t, max_size, accum_t, output_t>;
+  constexpr auto scan_op           = op_t{};
+  constexpr auto operator_identity = cub_operator_to_identity<accum_t, op_t>::value();
+  const int num_items              = GENERATE_COPY(take(3, random(1, max_size)));
```

Comment:
strong suggestion: I would suggest to make `num_items` a compile-time value. Otherwise, we'll be compiling `16` kernel instantiations, which is another huge multiplicative factor to the already extensive cross product of types and operators we test for. This will save us the `switch` statement:

I would declare 
```
using items_per_thread_list = c2h::enum_type_list<int, 1, 3, 15, 16>;
```
next to `integral_type_list`. 

Extend:
```
C2H_TEST("ThreadScanExclusive Integral Type Tests", "[scan][thread]", integral_type_list, cub_operator_integral_list, items_per_thread_list)
```

...and then:
```suggestion
  constexpr int num_items              = c2h::get<2, TestType>::value;
```